### PR TITLE
fix(DatePicker): fixes error state on focus

### LINKED ISSUE #548: DatePicker/DateRangePicker: fix error state on focus
### Steps to reproduce

1. Pass `withInput: true` and `inputOptions: {{ required: true }}` to DatePicker with no pre-defined `date`.
2. Focus on Input.

### Expected result

Error state should be shown on Input blur.

### Actual result

Error state is visible on Input focus.

### PR DESCRIPTION
**Pull request by riyalohia**
Closes #548

### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/organisms/datePicker/Trigger.tsx [hunk 1]
```diff
@@ -58,7 +56,8 @@ export const Trigger = (props: TriggerProps) => {
     });
   };
 
-  const showError = inputOptions.required && error && init;
+  const showError = inputOptions.error || (inputOptions.required && error && init);
+  const errorMessage = inputOptions.caption === undefined ? 'Invalid value' : inputOptions.caption;
```
Comment (aditya-kumawat) [inline core/components/organisms/datePicker/Trigger.tsx:60]: `const errorMessage = inputOptions.caption || 'Invalid value';`

File: core/components/organisms/dateRangePicker/SingleInputTrigger.tsx [hunk 2]
```diff
@@ -31,7 +31,8 @@ export const SingleInputTrigger = (props: TriggerProps) => {
   } = state;
 
   const mask = Utils.masks.rangeDate[inputFormat];
-  const showError = inputOptions.required && (startError || endError) && init;
+  const showError = inputOptions.error || (inputOptions.required && (startError || endError) && init);
+  const errorMessage = inputOptions.caption === undefined ? 'Invalid value' : inputOptions.caption;
```
Comment (aditya-kumawat) [inline core/components/organisms/dateRangePicker/SingleInputTrigger.tsx:35]: `const errorMessage = inputOptions.caption || 'Invalid value';`

File: core/components/organisms/dateRangePicker/Trigger.tsx [hunk 3]
```diff
@@ -127,8 +127,10 @@ export const Trigger = (props: TriggerProps) => {
   };
 
   const mask = Utils.masks.date[inputFormat];
-  const showStartError = startInputOptions.required && startError && init;
-  const showEndError = endInputOptions.required && endError && init;
+  const showStartError = startInputOptions.error || (startInputOptions.required && startError && init);
+  const showEndError = endInputOptions.error || (endInputOptions.required && endError && init);
+  const startErrorMessage = startInputOptions.caption === undefined ? 'Invalid value' : startInputOptions.caption;
```
Comment (aditya-kumawat) [inline core/components/organisms/dateRangePicker/Trigger.tsx:132]: `const startErrorMessage = startInputOptions.caption || 'Invalid value';`

File: core/components/organisms/dateRangePicker/Trigger.tsx [hunk 4]
```diff
@@ -127,8 +127,10 @@ export const Trigger = (props: TriggerProps) => {
...
+  const endErrorMessage = endInputOptions.caption === undefined ? 'Invalid value' : endInputOptions.caption;
```
Comment (aditya-kumawat) [inline core/components/organisms/dateRangePicker/Trigger.tsx:133]: `const endErrorMessage = endInputOptions.caption || 'Invalid value';`

**Review by aditya-kumawat — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**